bolt-simple-taproot.md

```diff
@@ -377,348 +352,276 @@ The Context column decodes as follows:
-
-The sending node SHOULD:
-
-  - use the `NonceGen` algorithm defined in `bip-musig2` to generate
-    `signing_musig2_pubnonce` and `verification_musig2_pubnonce` to ensure it
-    generates nonces in a safe manner.
-
-The receiving node MUST fail the channel if:
-
-  - the message doesn't include a `signing_musig2_pubnonce` and
-    `verification_musig2_pubnonce` value.
-
-  - a specified public nonce cannot be parsed as two compressed secp256k1
-    points
-
-##### Rationale
-
-For the initial Taproot channel type (`option_simple_taproot`), musig2 nonces
-must be exchanged in the first round trip (open->, <-accept) to ensure that the
-initiator is able to generate a valid musig2 partial signature at the next
-step once the transaction to be signed is fully specified. 
-
-The `signing_musig2_pubnonce` will be used to sign the _remote_ party's
-commitment transaction, while the `verification_musig2_pubnonce` will be used
-to verify incoming commitment signatures. At force close broadcast time, the
-verification nonce is then used to sign the local party's commitment
-transaction.
-
-We require two nonces, as there're actually two messages being signed: the
-local commitment and the remote commitment. If only one nonce was used, when a
-party went to sign and broadcast their local commitment, they would re-use a
-nonce, as the nonce was already used to sign the remote party's commitment.
+We add `option_taproot` to the set of defined channel types.
 
 #### `accept_channel` Extensions
 
 1. `tlv_stream`: `accept_channel_tlvs`
 2. types:
-    1. type: 2 (`verification_musig2_pubnonce`)
-    2: data:
-        * [`66*byte`:`nonces`]
-    1. type: 4 (`signing_musig2_pubnonce`)
-    2: data:
-        * [`66*byte`:`nonces`]
+   1. type: 4 (`musig2_nonce`)
+   2. data:
+      * [`66*byte`:`public_nonce`]
 
 ##### Requirements
 
 The sender:
 
- - MUST set `verification_musig2_pubnonce` and `signing_musig2_pubnonce` to the
-   `musig2` public nonce as specified by the `NonceGen` algorithm of
-   `bip-musig2`
+- MUST set `musig2_nonce` to the `musig2` public nonce used to sign local commitments as specified by the `NonceGen`
+  algorithm of `bip-musig2`.
 
- - Aggregate the received local+remote public nonces (into two combined nonces)
-   using the `NonceAgg` algorithm of `bip-musig2`.
-
-The receiver:
+The recipient:
 
- - MUST reject the channel if `signing_musig2_pubnonce` or
-   `verification_musig2_pubnonce` are not present.
+- MUST reject the channel if `musig2_nonce` is absent.
+- MUST store `musig2_nonce` until they send the `funding_created` message.
+  - SHOULD only store `musig2_nonce` in RAM, and not persist it to disk.
 
 #### `funding_created` Extensions
 
+1. `tlv_stream`: `funding_created_tlvs`
+2. types:
+   1. type: 2 (`partial_signature_with_nonce`)
+   2. data:
+      * [`98*byte`: `partial_signature || public_nonce`]
+   3. type: 4 (`musig2_nonce`)
+   4. data:
+      * [`66*byte`: `public_nonce`]
```

Comment:
> Is that really better than having her send it at the very start in open_channel? Then at this point, all nonces are already known and only a signature needs to be sent.

Also today after `accept_channel` is sent, all the necessaray fundign params have been exchanged (channel config effectively static). With this, you continue to send funding params even up until the very last message in the funding flow.

Reply:
> I would rather keep the simpler interpretation of nonces over that

So I think this is the main thing I'm contesting here: is sending nonces _later_ in the process actually simpler? Or is it just shifting things around w/ minimal gain?

As an example, in the new flow, Alice doesn't send her signing+verification nonce until the 3rd message is exchanged. But there's no reason why she can't just send that in `open_channel` since there're no hard dependancies on her generating the nonce (the new musig 1.0 mixes in the private key _always_ she gets randomness augmentation by default). 

I think the argument is stronger w/ the state machine changes though, since Bob doesn't need to cache Alice's "next" signing nonce, as it'll just be delivered to him w/ the next sig.

Reply:
So like you did, I also wanna break it down into the two components involved: one is sending the remote nonce as part of the partial signature, and the other is sending the local nonce just in time.

I don't think we have a disagreement regarding the utility of including the remote nonce with the partial signature, so really the discussion is about whether the first local nonce should be included in the open_channel message, or in funding_created. From an implementation perspective, there is, as you said, not much of a difference, because either way, the sender needs to cache the nonce until they receive a partial signature to verify the correctness. But there is a difference in the semantics.

Specifically, I believe that the nonce/partial signature dance is essentially a challenge response mechanism. Alice signals to Bob that she's ready for a signature, and here's the nonce that signature should aggregate against. When should that nonce be sent? In my opinion, when readiness for that signature is signaled, i. e. in the message that immediately precedes the partial signature being sent by the remote party.

Separately, Wilmer made the point that the open_channel message is really about setting up the channel parameters, and the nonce for the first signature really isn't a parameter of the channel.

So these are the reasons that we think it makes semantically more sense for the local nonce to be moved to the message that we moved them to, but again, as you pointed out, there's no big difference from the standpoint of actually needing to implement it — just from understanding the rationale behind why the implementation needs to be a certain way.

Reply:
> So I think this is the main thing I'm contesting here: is sending nonces later in the process actually simpler? Or is it just shifting things around w/ minimal gain?

It is definitely just shifting things around, but I think it has some merit. Not sending nonces upfront makes more logical sense in my head, as why would you send them before you know whether you'll ever get so far in the funding process that you'll actually need them?

Reply:
Good point! 

By the way, if we all agree that remote nonces should be made part of the partial signature, I think it would be somewhat less confusing to external readers if we were referring to the nonce in singular, given that it's really only the local nonce that has its own dedicated TLV field.